Multisig/UI/Settings/OwnerKeyManagement/GenerateOwnerKey/OnboardingGenerateKeyViewController.swift

```diff
@@ -73,14 +73,55 @@ class OnboardingGenerateKeyViewController: AddKeyOnboardingViewController {
     func startBackupFlow() {
         let backupController = BackupController(showIntro: true, seedPhrase: mnemonic)
         backupController.onComplete = { [weak self] in
-            self?.showKeyDetails()
+            guard let self = self else { return }
+            if self.canAddKeyAsOwner() {
+                self.startAddKeyAsOwner()
+            } else {
+                self.showKeyDetails()
+            }
         }
         backupController.onCancel = { [weak self] in
-            self?.showKeyDetails()
+            guard let self = self else { return }
+            self.dismiss(animated: true) {
+                if self.canAddKeyAsOwner() {
+                    self.startAddKeyAsOwner()
+                } else {
+                    self.showKeyDetails()
+                }
+            }
         }
         show(backupController, sender: self)
     }
 
+    func canAddKeyAsOwner() -> Bool {
+        guard let safe = try? Safe.getSelected() else { return false }
+        if !safe.isReadOnly {
+            return true
+        } else {
+            return true
+        }
```

Comment:
it's a strange if-else returning true in all cases. Maybe just return the value of `!safe.isReadOnly`?